tools/test/h5dump/h5dumpgentest.c

```diff
@@ -11521,12 +11629,18 @@ gent_onion_1d_dset(void)
 
     /* Setup */
     onion_info.backing_fapl_id = H5Pcreate(H5P_FILE_ACCESS);
+    /* Force onion files to be generated via Native connector */
+    if (H5Pset_vol(onion_info.backing_fapl_id, H5VL_NATIVE, NULL) < 0)
```

Comment:
Forcing the native VOL is not the way to go about this, especially since this could be performed on various passthrough VOL connectors which should generally work fine. Skipping generation of certain test files for some connectors could potentially be acceptable, but I'd much rather that skipping be done at the build system level or higher rather than here. I'm thinking more in the direction of "all test files should always be generated, regardless of VOL connector". It's ok if some files can't be generated due to something a connector doesn't support, like going through the VFL interface in this case, but we shouldn't try to force these situations.